### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by migo1**
### MILESTONES

- Use API that allows to get numeric values based on a parameter
- Use design guidelines including:
           - Colors (select one main color for your website).
           - Typography: font face, size, and weight.
           - Layout: composition and space between elements.
          
### Interactions

**Home page**

- When the page loads, the web app shows the list of categories that could be filtered out by some parameter, for example by the category name.
- Along with the category name, you will display some numeric values per category that come from the API.
- When the user clicks (or taps) on a category item, the application navigates to the details page.

**Details page**

- In the details page, the web app retrieves data from the API to show detailed data in the category.
- When the user clicks on the "Back" button (<), the user navigates to the home page.

**Testing requirements**

- Create unit tests for pure functions in your Redux code.
- Create integration tests for your application using the React Testing Library.
- You may need to mock the access to the API so that your tests don't send actual requests.
- You may need to mock the connection to the Redux Store.

Technical requirements

- The project is a single-page application (SPA) built with React and Redux.
- The data retrieved from the API should be stored in the Redux store.
- You should filter the data that you retrieve from the API using a Filter stateless component.
- Every page (the main page and the pages for each item) should have a unique route within the SPA.
- The project is be deployed and accessible online using [Netlify](https://www.netlify.com/) .
- Add a link to the online version of your app to your README file.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,172 @@
...
+
+Give a ⭐️ if you like this project!
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- ACKNOWLEDGEMENTS -->
+
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+
+- Thanks to all Microverse community
```
Comment (Ibizugbe) [inline README.md]: As required in this milestone, I suggest you add an acknowledgment to the author of the design, [Nelson Sakwa on Behance](https://www.behance.net/sakwadesignstudio). In fulfillment of the requirement of the [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/)😉😉.

image

File: README.md [hunk 2]
```diff
@@ -1 +1,172 @@
...
+- **Statistics of each NBA team** 
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- LIVE DEMO -->
+
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://dev--marvelous-mandazi-c7fe36.netlify.app/)
+- [video walkthrough Link](https://www.loom.com/share/30fdc28e49524da69ffb00842d47eb26?sid=f617c166-ce16-4cd9-8177-1215b94615a4)
```
Comment (Ibizugbe) [inline README.md]: Nice Job so far, your video presentation looks great👍🏼. But you seem to have forgotten to make mention of a piece of code or feature in the application that you implemented and you are proud of😉. Kindly do well to talk about that as this is a requirement in your video presentation👍🏼.

image

File: src/App.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        <Nabar />
+        <Routes>
+          <Route path="/" element={<Teams />} />
+          <Route path="/:teamId/statistics" element={<TeamStats />} />
+        </Routes>
+      </Router>
+    </div>
+  );
+}
+
```
Comment (Ibizugbe) [inline src/App.js]: Nice job so far, Your project looks great👍🏼. Please let's try to make it responsive as it currently looks different from the given design on the mobile screen😉.

Your project
image

design
image


Kindly note that this is the same for all pages, as the mobile view wasn't considered😉. Do well to make the necessary changes to make your project match the design👍🏼.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @migo1 👋🏼👋🏼👋🏼, 

require_changes


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Project has been deployed using netlify and the deployed version works just fine👍🏼.
- [x] The file structure of the project looks great👍🏼.
- [x] The project contains a Home page and a Detail page👍🏼.
- [x] It's possible to navigate the detail page by clicking the individual category item👍🏼.
- [x] The project is a single page application (SPA) using `react-router-dom` to serve each page without reloading👍🏼.
- [x] The project uses Redux for state management and data retrieved from the API are stored in the Redux store correctly👍🏼.
- [x] Unit tests and integration tests has been written for components and some pure functions in redux👍🏼.


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by migo1** [comment]
@Ibizugbe noted, thanks for the review and corrections

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @migo1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

👍 You did well to fix all the requirements requested by the previous reviewer
👏 The pages are responsive(mobile)

## Optional suggestions

- [ ] You used the wrong workflow, all the work would have been done in a feature branch merging into the development branch.

gitflow

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**